src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -237,6 +239,69 @@ public Locatable getLocation() {
     }
 
+    /**
+     * Fixes the quality of all the elements that come from an overlapping pair in the same way as
+     * samtools does {@see tweak_overlap_quality function in
+     * <a href="https://github.com/samtools/htslib/blob/master/sam.c">samtools</a>}.
+     *
+     * Setting the quality of one of the bases to 0 effectively removes the redundant base for
+     * calling. In addition, if the bases overlap we have increased confidence if they agree (or
+     * reduced if they don't). Thus, the algorithm proceeds as following:
+     *
+     * 1. If the bases are the same, the quality of the first element is the sum of both qualities
+     *    and the quality of the second is reduced to 0.
+     * 2. If the bases are different, the base with the highest quality is reduced with a factor of
+     *    0.8, and the quality of the lowest is reduced to 0.
+     *
+     */
+    public void fixOverlaps() {
+        final FragmentCollection<PileupElement> fragments = FragmentCollection.create(this);
+        fragments.getOverlappingPairs().stream()
+                .forEach(
+                        elements ->  fixPairOverlappingQualities(elements.get(0), elements.get(1))
+                );
+    }
+
+    /**
+     * Fixes the quality of two elements that come from an overlaping pair in the same way as
+     * samtools does {@see tweak_overlap_quality function in
+     * <a href="https://github.com/samtools/htslib/blob/master/sam.c">samtools</a>}.
+     * The only difference with the samtools API is the cap for high values (Byte.MAX_VALUE).
+     */
+    @VisibleForTesting
+    static void fixPairOverlappingQualities(final PileupElement firstElement,
+                                            final PileupElement secondElement) {
+        // only if they do not represent deletions
+        if (!secondElement.isDeletion() && !firstElement.isDeletion()) {
+            final byte[] firstQuals = firstElement.getRead().getBaseQualities();
+            final byte[] secondQuals = secondElement.getRead().getBaseQualities();
+            if (firstElement.getBase() == secondElement.getBase()) {
+                // if both have the same base, extra confidence in the firts of them
+                firstQuals[firstElement.getOffset()] =
+                        (byte) (firstQuals[firstElement.getOffset()] + secondQuals[secondElement
+                                .getOffset()]);
+                // cap to maximum byte value
+                if (firstQuals[firstElement.getOffset()] < 0) {
+                    firstQuals[firstElement.getOffset()] = Byte.MAX_VALUE;
+                }
+                secondQuals[secondElement.getOffset()] = 0;
+            } else {
+                // if not, we lost confidence in the one with higher quality
+                if (firstElement.getQual() >= secondElement.getQual()) {
+                    firstQuals[firstElement.getOffset()] =
+                            (byte) (0.8 * firstQuals[firstElement.getOffset()]);
```

Comment:
A suggestion for make this better will be to add a `double` parameter to downgrade the confidence of a mismatch. Like that I can use it with 0.8 to mimic samtools behaviour (and change it if the samtools method is updated), and other API users could use more reasonable values (e.g. PCR error rate correction and/or just set to 0 to do not take it into account).